superset-frontend/src/dashboard/components/DashboardBuilder/DashboardContainer.tsx

```diff
@@ -85,9 +86,17 @@ const DashboardContainer: FC<DashboardContainerProps> = ({ topLevelTabs }) => {
   const directPathToChild = useSelector<RootState, string[]>(
     state => state.dashboardState.directPathToChild,
   );
-  const chartIds = useSelector<RootState, number[]>(state =>
-    Object.values(state.charts).map(chart => chart.id),
+  const isEditMode = useSelector<RootState, boolean>(
+    state => state.dashboardState.editMode,
   );
+  const charts = useSelector<RootState, Chart[]>(state =>
```

Comment:
It could probably be further optimised, so that we don't repeat some operations, but the general idea is to return values/objects from useSelector that change scarcely

Reply:
I meant that we should get rid if this selector altogether

Reply:
gotcha, lets merge and I'll do it when I work on perf